widget/entry_test.go

```diff
@@ -1304,3 +1304,20 @@ func TestEntry_EraseEmptySelection(t *testing.T) {
 	assert.Equal(t, -1, a)
 	assert.Equal(t, -1, b)
 }
+
+func TestPasswordEntry_Reveal(t *testing.T) {
+	entry := NewPasswordEntry()
+
+	test.Type(entry, "Hié™שרה")
+	assert.Equal(t, "Hié™שרה", entry.Text)
+	assert.Equal(t, "*******", entryRenderTexts(entry)[0].Text)
+
+	// Reveal password that should be obfuscated until it is refreshed
+	entry.Password = false
+	assert.Equal(t, "Hié™שרה", entry.Text)
```

Comment:
I don't think that we should check this line and the next again.
Refresh should guarantee an update - but not calling it is not a guarantee of no update, so this check is not part of the API contract

Reply:
Removed in 43f8def